### PR TITLE
PIM-5956: Add command to query and bulk delete products

### PR DESCRIPTION
**Pull request by solivier**
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | No
| Added Behats                      | No
| Changelog updated                 | No
| Review and 2 GTM                  | No

This command allow to query with the PQB and delete in bulk the selected products

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+use Pim\Bundle\CatalogBundle\Query\ProductQueryBuilderInterface;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Query and delete bulk products
```
Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php]: Query and delete products by bulk

**Comment by damien-carcel** [comment]
Please add a description to the PR. Why is this command needed?

File: src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+ *
+ * @author    Olivier Soulet <olivier.soulet@akeneo.com>
+ * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class QueryAndDeleteProductCommand extends ContainerAwareCommand
+{
+    /* @var integer */
```
Comment (damien-carcel) [inline src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php]: No PHPDoc on constants

File: src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+        if (!$helper->ask($input, $output, $question)) {
+            return;
+        }
+
+        $paginator = $this->getPaginatorFactory()->createPaginator($cursor);
+
+        foreach ($paginator as $productsPage) {
+            $this->getProductRemover()->removeAll($productsPage);
+            $output->write(sprintf('%s products deleted ...\n', count($productsPage)));
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php]: `\n` doesn't work with simple quote

File: src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                'operator' => '=',
+                'value'    => '100',
+                'locale'   => 'en_US',
+                'scope'    => 'print'
+            ]
+        ];
+
+        $this
+            ->setName('pim:product:remove')
+            ->setDescription('Query products and deletes them')
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php]: `Delete products` ?

**Comment by juliensnz** [comment]
GTM, nice work :)

File: src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php (hunk 4 above)
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php]: You need to select which products you want to delete so you have to query them with pqb filter

File: src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+            return;
+        }
+
+        $paginator = $this->getPaginatorFactory()->createPaginator($cursor);
+
+        foreach ($paginator as $productsPage) {
+            $this->getProductRemover()->removeAll($productsPage);
+            $output->write(sprintf("%s products deleted ...\n", count($productsPage)));
+            $this->getProductDetacher()->detachAll($productsPage);
+        }
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php]: add a message at the end the process please

Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Command/QueryAndDeleteProductCommand.php]: done

**Comment by damien-carcel** [comment]
GTM

**Pull request merged**